### PR TITLE
[#2473] refactor: enable UnusedVariable error-prone

### PR DESCRIPTION
**Pull request by coolderli**
### What changes were proposed in this pull request?

- enable  UnusedVariable error-prone

### Why are the changes needed?

Fix: #2473 

### Does this PR introduce _any_ user-facing change?

- no

### How was this patch tested?

- `./gradlew build`

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -350,6 +351,7 @@ subprojects {
         "UnnecessaryTypeArgument",
         "UnusedAnonymousClass",
         "UnusedCollectionModifiedInPlace",
+        "UnusedVariable",
```
Comment (qqqttt123) [inline build.gradle.kts:354]: Could you let the Error-prone code follow the alphabet order? Such as `EqualsGetClass`, `DefaultCharset`, `InlineFormatString`.

File: catalogs/bundled-catalog/src/main/java/com/datastrato/gravitino/catalog/common/ClassProvider.java [hunk 2]
```diff
@@ -36,6 +36,7 @@
  * Therefore, this module is used to store the required classes for the query engine's
  * functionality.
  */
+@SuppressWarnings("UnusedVariable")
```
Comment (qqqttt123) [inline catalogs/bundled-catalog/src/main/java/com/datastrato/gravitino/catalog/common/ClassProvider.java:39]: Could you explain why we add this annotation?

File: clients/client-java/src/main/java/com/datastrato/gravitino/client/HTTPClient.java [hunk 3]
```diff
@@ -634,11 +634,10 @@ public void close() throws IOException {
   /**
    * Creates a new instance of the HTTPClient.Builder with the specified properties.
    *
-   * @param properties A map of properties (key-value pairs) used to configure the HTTP client.
    * @return A new instance of HTTPClient.Builder with the provided properties.
    */
-  public static Builder builder(Map<String, String> properties) {
-    return new Builder(properties);
+  public static Builder builder() {
```
Comment (qqqttt123) [inline clients/client-java/src/main/java/com/datastrato/gravitino/client/HTTPClient.java:639]: I will prefer reserving this.

File: build.gradle.kts (hunk 1 above)
Comment (coolderli) [reply build.gradle.kts]: Done

File: clients/client-java/src/main/java/com/datastrato/gravitino/client/HTTPClient.java (hunk 3 above)
Comment (coolderli) [reply clients/client-java/src/main/java/com/datastrato/gravitino/client/HTTPClient.java]: Done

File: catalogs/bundled-catalog/src/main/java/com/datastrato/gravitino/catalog/common/ClassProvider.java (hunk 2 above)
Comment (qqqttt123) [reply catalogs/bundled-catalog/src/main/java/com/datastrato/gravitino/catalog/common/ClassProvider.java:39]: cc @yuqi1129 Could you give some advice?

Comment (yuqi1129) [reply catalogs/bundled-catalog/src/main/java/com/datastrato/gravitino/catalog/common/ClassProvider.java:39]: It's a class that holder class that may be needed by query engines like `Spark` and `Trino` and won't be used directly in the project.

image

**Review by qqqttt123 — APPROVED** [review]
LGTM, thanks @coolderli

**Comment by qqqttt123** [comment]
@coolderli Could you fix the issue of CI?

**Pull request merged**